### PR TITLE
[ML] Add latest search interval to datafeed stats

### LINKED ISSUE #82405: [ML] Add latest search start and end time in datafeed stats
When a datafeed is running performing searches that scan a long time period (particularly in lookbacks)
it would be very useful to report where the search is in the stats API. For example, in the scenario where the
datafeed is going through an extended period without any data, it could seem like the job does nothing. But
if the user could see the search time progressing they would understand that the job is progressing fine.

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
This commit adds `search_interval` to the datafeed stats API
`running_state` object. When the datafeed is running, it reports
the last search interval that was searched. It is useful to
understand the point in time where the datafeed is currently
searching.

Closes #82405

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/SearchInterval.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public record SearchInterval(long startMs, long endMs) implements ToXContentObject, Writeable {
```
Comment (benwtrent) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/SearchInterval.java:19]: :shipit: 🚀

**Review by benwtrent — APPROVED** [review]

File: docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc [hunk 2]
```diff
@@ -112,6 +112,17 @@ has no configured `end` time.
 (boolean) Indicates whether the {dfeed} has finished running on the available
 past data. For {dfeeds} without a configured `end` time, this means that
 the {dfeed} is now running on "real-time" data.
+
+`search_interval`:::
+(Optional, object) An object that provides the latest time interval the {dfeed}
+has searched.
```
Comment (szabosteve) [inline docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:118]:
```suggestion
(Optional, object) Provides the latest time interval the {dfeed} has searched.
```

File: docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc [hunk 3]
```diff
@@ -112,6 +112,17 @@ has no configured `end` time.
...
++
+[%collapsible%open]
+=====
+`start_ms`: The start time as an epoch in milliseconds.
+`end_ms`: The end time as an epoch in milliseconds.
```
Comment (szabosteve) [inline docs/reference/ml/anomaly-detection/apis/get-datafeed-stats.asciidoc:123]:
```suggestion
`start_ms`::::
The start time as an epoch in milliseconds.

`end_ms`::::
 The end time as an epoch in milliseconds.
```

**Review by szabosteve — APPROVED** [review]
Left two comments, the second one should fix the formatting issue. Otherwise, LGTM! Thanks!

**Comment by dimitris-athanasiou** [comment]
@elasticmachine update branch

**Pull request merged**